docs: Update relationship docs [DHIS2-19169] (#1516)
dhis2/dhis2-docs: src/user/relationship-model.md

> # Relationship model { #relationship_model }

Reviewer: I think we should add a diagram to the model. You adding a diagram to the PR made it easier to understand than all of the text/explanations you gave before. I might not be the only person which needs some visual 😅 

Something like https://graphviz.org/ or https://d2lang.com/ is useful so anyone can adapt and regenerate the image.

Author: I agree, I will have a look at this tomorrow but I will probably ask for some help as I am not sure how this work in the markdown docs.
In the meantime, if the rest is ok I would already merge it and move the ticket to test.

<!--DHIS2-SECTION-ID:relationship_model-->

A relationship represents a link between two entities in the Tracker-model. A
relationship is considered data in DHIS2 and is based on a Relationship Type,

> similar to how a Tracked Entities are based on a Tracked Entity Type.

Reviewer 2: either `how a Tracked entity is based` or `how Tracked entities are based`

Reviewer: I am for `tracked entities` as said in above comment

> Relationships always include two entities, and these entities can include Tracked

Reviewer: ```suggestion
Relationships always connect two entities, and these entities can include Tracked
```

or `link` not sure which word we want to stick to

Entities, Enrollments and Events, and any combination of these. Note

> that not all of these combinations are available in the current apps.

Reviewer 2: ```suggestion
that all of these combinations are not currently available through the User Interface.
```

> Relationships between entities can be classified as either unidirectional or bidirectional, with each type requiring different levels of access for creation.

Reviewer: Please set a max width of 100 (at least for markdown) in your editor

* Unidirectional relationships require the user to have:
    * Write access to the "from" entity
    * Read access to the "to" entity
* Bidirectional relationships require write access to both the "from" and "to" entities.

> When retrieving relationships from a [tracker exporter endpoint](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/tracker.html#webapi_tracker_export), only those originating from the requested entity are shown.

Reviewer: Use anchors so the links keep working on any doc branch/version

> This means that the entity is either part of a bidirectional relationship or the "from" entity in a unidirectional relationship.

Reviewer: Keep it in one paragraph as this explains `originating` above

set before it can be created. For example, if your relationship is between a
mother and a child, both constraints would have their required Tracked Entity
Type set to Person and could optionally set the Enrollment to Maternal Health
Program and Child Program respectively. This way, only Tracked Entities
who are of type Person and who is enrolled in the required program is allowed to
be included in these relationships.

In addition to the constraints defined by a Relationship Type, each relationship can be configured as bidirectional, true or false.

> * If the property is set to false, the relationship is treated as unidirectional.

Reviewer: ```suggestion
* If the property is set to false, the relationship is unidirectional.
```

* If set to true, the relationship is bidirectional.

> As mentioned earlier, this setting impacts the required access levels for creating the relationship, as well as the relationships displayed when exporting an entity that is part of the relationship.

Reviewer: Please link to the section you mention

```suggestion
As mentioned earlier, this setting impacts the required access levels for creating the relationship, and the relationships displayed when exporting an entity that is part of the relationship.
```

who are of type Person and who is enrolled in the required program is allowed to
be included in these relationships.

In addition to the constraints defined by a Relationship Type, each relationship can be configured as bidirectional, 
true or false.

* If the property is set to false, the relationship is unidirectional.
* If set to true, the relationship is bidirectional.
As mentioned earlier in [here](#relationship_model), this setting impacts the required access levels for creating 

> * the relationship, and the relationships displayed when exporting an entity that is part of the relationship.

Reviewer: ```suggestion
the relationship, and the relationships displayed when exporting an entity that is part of the relationship.
```

either no bullets or at least two 😁

One important thing to note about bidirectional relationships, are that the
"from" and "to" sides are still significant in the database, meaning each entity
must match the constraint for that side. However, from a user perspective, which
side each entity is stored as is insignificant.

> ```dot

Reviewer: accident
